knotx-core/src/main/java/io/knotx/dataobjects/KnotError.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2016 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.knotx.dataobjects;
+
+import com.google.common.base.Objects;
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
+
+@DataObject(inheritConverter = true)
+public class KnotError {
+
+  private static final String CODE_KEY = "_CODE";
+  private static final String MESSAGE_KEY = "_MESSAGE";
+
+  private String code;
+  private Object message;
```

Comment:
What is the reason for `message` to be the `Object` not `String`?

Reply:
I've had a request to make it an Object to have the ability to store structured data. Opting to keep it this way